src/preprocessor.rs

```diff
@@ -400,6 +450,103 @@ mod test {
         assert_eq!(in_ns, 42);
     }
 
+    fn textual_prefix(len: usize) -> String {
+        format!("{} {}", len, String::from_utf8(vec![b'O'; len]).unwrap())
+    }
+
+    use proptest::prelude::*;
+
+    // generate multiple chopped length-prefixed strings
+    fn multiple_textual_lengths(max_elements: usize) -> BoxedStrategy<(Vec<usize>, Vec<String>)> {
+        proptest::collection::vec(".+", 1..max_elements) // generator for Vec<String> of arbitrary strings, maximum length of vector: `max_elements`
```

Comment:
Maybe, if we change the regex in the string to `"+*"` we will also get empty strings and could test the preprocessor against that?